### ISSUE TITLE
Prometheus Integration

### ISSUE THREAD

**Issue by alexjustesen** [alexjustesen/speedtest-tracker#53]
### Discussed in https://github.com/alexjustesen/speedtest-tracker/discussions/24

<div type='discussions-op-text'>

<sup>Originally posted by **Nexxed** September 19, 2022</sup>
This is an open issue on the repo of the previous project and it would be cool to both support InfluxDB and Prometheus (for those that use one or the other):
https://github.com/henrywhitaker3/Speedtest-Tracker/issues/658</div>

**Comment by svenvg93** [comment]
@alexjustesen Before I start trying to build this, do you still want it? Or do you plan only to expose data via the API? 
As for InfluxDB it might be easier to have the API send a json back which can be pushed into InfluxDB via telegram. Just to simplify the data export in the app. just my two cents.

**Comment by svenvg93** [comment]
Perfect, no worries!

**Comment by phyzical** [comment]
just stumbled across this as a workaround in the mean time https://github.com/prometheus/influxdb_exporter

can confirm it seems to do the job

**Comment by jiriteach** [comment]
> just stumbled across this as a workaround in the mean time https://github.com/prometheus/influxdb_exporter
> 
> can confirm it seems to do the job

Yup - works except, am seeing data being wiped on the metrics endpoint. Have you seen this? ~ 10/15 mins after a speedtest - data on the /metrics endpoint seems to be wiped.

**Comment by phyzical** [comment]
Yeah it's by design, your actual Prometheus instance ends up keeping it stored if that makes sense. So exporter -> prom -> grafana

**Comment by jiriteach** [comment]
> Yeah it's by design, your actual Prometheus instance ends up keeping it stored if that makes sense. So speed test -> exporter -> prom -> grafana

Any idea how to make the Prometheus instance keep the data? Finding mine is clearing it. Because prom is checking the exporter and the exporter is clear - it's showing blank results after 10-15 mins.

**Comment by phyzical** [comment]
you need to have an actual prometheus instance hit the exporter, that prometheus instance is what stores it.

If your main instance is blatting your data youll need to google fu prometheus

**Comment by phyzical** [comment]
@jiriteach fyi i found there is this flag `--influxdb.sample-expiry 30m` be default its only 5 mins

**Comment by jiriteach** [comment]
> @jiriteach fyi i found there is this flag `--influxdb.sample-expiry 30m` by default its only 5 mins

Thanks

**Comment by sunjam** [comment]
Found this [grafana dashboard](https://github.com/t4ir1/speedtest-dashboard/tree/main)
